src/main/java/ch/njol/util/coll/iterator/CombinedIterator.java

```diff
@@ -10,8 +10,9 @@
  * <p>
  * Elements are removable from this iterator if the source iterables support element removal, unless removal is blocked on creation.
  * 
- * @author Peter Güttinger
+ * @deprecated use multiple {@link Iterator}s for finer control
```

Comment:
There is https://guava.dev/releases/snapshot-jre/api/docs/com/google/common/collect/Iterators.html#concat(java.util.Iterator...)